tests/msc3930_test.go

```diff
@@ -0,0 +1,182 @@
+		// Request each of the push rule IDs defined by MSC3930 and verify their structure.
+
+		// This push rule silences all poll responses.
+		pollResponseRule := alice.GetPushRule(t, "global", "override", pollResponsePushRuleID)
+		must.MatchGJSON(
+			t,
+			pollResponseRule,
+			match.JSONKeyEqual("actions", []string{}),
+			match.JSONKeyEqual("default", true),
+			match.JSONKeyEqual("enabled", true),
+			// There should only be one condition defined for this type
+			match.JSONKeyEqual("conditions.#", 1),
+			// Check the contents of the first (and only) condition
+			match.JSONKeyEqual("conditions.0.kind", "event_match"),
+			match.JSONKeyEqual("conditions.0.key", "type"),
+			match.JSONKeyEqual("conditions.0.pattern", "org.matrix.msc3381.poll.response"),
+		)
+
+		// This push rule creates a sound when a poll is started in a one-to-one room.
+		pollStartOneToOneRule := alice.GetPushRule(t, "global", "underride", pollStartOneToOneRuleID)
+		must.MatchGJSON(
+			t,
+			pollStartOneToOneRule,
+			// Check that the appropriate actions are set for this rule
+			match.JSONKeyEqual("actions.#", 2),
+			match.JSONKeyEqual("actions", []interface{}{
+				"notify",
+				map[string]interface{}{"set_tweak": "sound", "value": "default"},
+			},
+			),
+			match.JSONKeyEqual("default", true),
+			match.JSONKeyEqual("enabled", true),
+			// There should two conditions defined for this type
+			match.JSONKeyEqual("conditions.#", 2),
+			// Check the condition that requires a room between two users
+			match.JSONKeyEqual("conditions.#(kind==\"room_member_count\").is", "2"),
+			// Check the condition that requires a poll start event
+			match.JSONKeyEqual("conditions.#(kind==\"event_match\").key", "type"),
+			match.JSONKeyEqual("conditions.#(kind==\"event_match\").pattern", "org.matrix.msc3381.poll.start"),
+		)
+
+		// This push rule creates a sound when a poll is ended in a one-to-one room.
+		pollEndOneToOneRule := alice.GetPushRule(t, "global", "underride", pollEndOneToOneRuleID)
+		must.MatchGJSON(
+			t,
+			pollEndOneToOneRule,
+			// Check that the appropriate actions are set for this rule
+			match.JSONKeyEqual("actions.#", 2),
+			match.JSONKeyEqual("actions", []interface{}{
+				"notify",
+				map[string]interface{}{"set_tweak": "sound", "value": "default"},
+			},
+			),
+			match.JSONKeyEqual("default", true),
+			match.JSONKeyEqual("enabled", true),
+			// There should two conditions defined for this type
+			match.JSONKeyEqual("conditions.#", 2),
+			// Check the condition that requires a room between two users
+			match.JSONKeyEqual("conditions.#(kind==\"room_member_count\").is", "2"),
+			// Check the condition that requires a poll start event
+			match.JSONKeyEqual("conditions.#(kind==\"event_match\").key", "type"),
+			match.JSONKeyEqual("conditions.#(kind==\"event_match\").pattern", "org.matrix.msc3381.poll.end"),
+		)
+
+		// This push rule creates a sound when a poll is started in any room.
```

Comment:
Does the push rule really create a sound when there's no `{"set_tweak": "sound", "value": "default"}` action? I'm not familiar with how they work.

Reply:
Oop, no, that's a copy/paste error. Good spot!